reasoner/planner/ConjunctionStreamPlan.java

```diff
@@ -0,0 +1,337 @@
+                //  use resolvableOutputs instead of rightOutputs because this node has to do the job of the parent as well - joining the identifiers
+                Set<Retrievable> resolvableOutputs = VariableSets.difference(requiredOutputs, variableSets.extendOutputWith);
+                return new ResolvablePlan(subConjunction.get(0), variableSets.rightInputs, variableSets.extendOutputWith, resolvableOutputs);
+            } else {
+                Pair<List<Resolvable<?>>, List<Resolvable<?>>> divided = divide(subConjunction);
+                VariableSets variableSets = VariableSets.determineVariableSets(divided.first(), divided.second(), availableInputs, requiredOutputs);
+                ConjunctionStreamPlan leftPlan = buildRecursivePrefix(divided.first(), variableSets.leftIdentifiers, variableSets.leftOutputs);
+                ConjunctionStreamPlan rightPlan = buildRecursiveSuffix(divided.second(), variableSets.rightInputs, variableSets.rightOutputs);
+
+                return new CompoundStreamPlan(list(leftPlan, rightPlan), variableSets.identifiers, variableSets.extendOutputWith, requiredOutputs);
+            }
+        }
+
+        public ConjunctionStreamPlan buildRecursiveSuffix(List<Resolvable<?>> subConjunction, Set<Retrievable> availableInputs, Set<Retrievable> requiredOutputs) {
+            // We should be able to call buildRecursivePrefix and get the same answer.
+            // If we did everything right, we should have an easy build.
+            if (subConjunction.size() == 1) {
+                VariableSets variableSets = VariableSets.determineVariableSets(list(), subConjunction, availableInputs, requiredOutputs);
+                //  use resolvableOutputs instead of rightOutputs because this node has to do the job of the parent as well - joining the identifiers
+                Set<Retrievable> resolvableOutputs = VariableSets.difference(requiredOutputs, variableSets.extendOutputWith);
+                return new ResolvablePlan(subConjunction.get(0), variableSets.rightInputs, variableSets.extendOutputWith, resolvableOutputs);
+            } else {
+                List<Resolvable<?>> suffix = subConjunction.subList(1, subConjunction.size());
+                VariableSets variableSets = VariableSets.determineVariableSets(subConjunction.subList(0, 1), subConjunction.subList(1, subConjunction.size()), availableInputs, requiredOutputs);
+                ConjunctionStreamPlan leftPlan = new ResolvablePlan(subConjunction.get(0), variableSets.leftIdentifiers, set(), variableSets.leftOutputs);
+                ConjunctionStreamPlan rightPlan = buildRecursiveSuffix(suffix, variableSets.rightInputs, variableSets.rightOutputs);
+                return new CompoundStreamPlan(list(leftPlan, rightPlan),
+                        variableSets.identifiers, variableSets.extendOutputWith, requiredOutputs);
+            }
+        }
```

Comment:
Building the suffix should be easy because it satisfies the structure described in the PR description.